app/controllers/katello/api/v2/upstream_subscriptions_controller.rb

```diff
@@ -22,9 +22,12 @@ class Api::V2::UpstreamSubscriptionsController < Api::V2::ApiController
     param :quantities_only, :bool, desc: N_("Only returns id and quantity fields")
     param :attachable, :bool, desc: N_("Return only subscriptions which can be attached to the upstream allocation")
     def index
-      pools = UpstreamPool.fetch_pools(upstream_pool_params.to_h)
+      index_params = upstream_pool_params
+      pools = UpstreamPool.fetch_pools(index_params)
+      page = index_params[:page] || 1
+
       collection = scoped_search_results(
-        pools[:pools], pools[:pools].count, pools[:total], params[:page], params[:per_page], nil)
+        pools[:pools], pools[:subtotal], pools[:total], page, index_params[:per_page], nil)
```

Comment:
Took the liberty of putting a :subtotal key on the pool response for cleanliness & clarity